test/functional/tui/child_session.lua

```diff
@@ -1,3 +1,9 @@
+-- XXX: This is a hack: we spawn a child `nvim` inside :terminal and use
+-- jobsend() to send raw bytes. This allows us to test the tui/input.c module,
+-- which does have a public API. But this means that the test session methods
+-- (eval(), request(), ... from helpers.lua) operate on the _host_ session,
+-- _not_ the embedded session.
```

Comment:
This is very confusing for people trying to follow the tests. I hope to improve this some more, but for now I renamed `terminal/helpers.lua` to `tui/child_session.lua`.